youtube_dl/extractor/thisvid.py

```diff
@@ -0,0 +1,111 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import json
+import re
+
+from .common import InfoExtractor
+from .openload import PhantomJSwrapper
+
+
+class ThisVidIE(InfoExtractor):
+    _VALID_URL = r'https?://(www\.)?thisvid\.com/videos/(?P<id>[A-Za-z0-9-]+)'
+
+    _TEST = {
+        'url': 'https://thisvid.com/videos/madonna-show-in-sexy-underwear/',
+        'md5': '48e38730d38394c6e9f1cce66fb04c6e',
+        'info_dict': {
+            'id': '829503',
+            'display_id': 'madonna-show-in-sexy-underwear',
+            'ext': 'mp4',
+            'title': 'Madonna show in sexy underwear',
+            'thumbnail': r're:^https?://.*preview\.mp4\.jpg$',
+            'uploader_id': 'Mike_Hunt',
+            'uploader_url': 'https://thisvid.com/members/584768',
+            'age_limit': 18,
+        }
+    }
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+
+        # The webpage contains a raw piece of javascript which creates a
+        # variable called flashvars used by the player to update the webpage.
+        #
+        # Running the javascript fixes the URL in the static HTML which is
+        # broken, and updates the flashvars variable with the new info.
+        jscode = """
+        function checkFlashVars() {
+            thumbnail_url = page.evaluate(function() {
+                return flashvars['preview_url']
+            });
+            fixed_url = page.evaluate(function() {
+                return flashvars['video_url']
+            });
+            console.log('---'); // ensure any errors appear above where we are
+            console.log(fixed_url);
+            console.log(thumbnail_url);
+            saveAndExit();
+        }
+        checkFlashVars();"""
+        phantom = PhantomJSwrapper(self, required_version='2.0')
+        webpage, output = phantom.get(url, html=webpage, jscode=jscode)
+        video_url, thumbnail_url = \
+            output.split("---", 2)[1].strip().split("\n", 2)
+        assert not video_url.startswith('function'), "old flashvars found"
+        video_url = video_url.split("?", 2)[0].strip("/")
```

Comment:
What's the point of this? Remove.

Reply:
The URL sometimes has extra stuff at the end. I can refactor this as part of the change above.

Reply:
Added comment